pygfunction/api.py

```diff
@@ -0,0 +1,141 @@
+
+    @staticmethod
+    def setup_list(check_data: FloatOrList, expected_size: int):
+        """
+        If the check_data argument is a float, this simply returns a list of expected size, filled with that value.
+        If the check_data argument is a list, this validates the size of the list and returns it.
+
+        Parameters
+        ----------
+        check_data: FloatOrList
+            Data or list of parameters to use
+        expected_size: int
+            Expected size of list
+
+        Returns
+        ----------
+        list of floats filled with correct values
+        """
+        if isinstance(check_data, list):
+            if len(check_data) != expected_size:
+                raise ValueError(f"Expected list of length {expected_size}, but got {len(check_data)}.")
+            return check_data
+        return [check_data] * expected_size
+
+    def as_config(self):
+        """"""
+        all_bh_configs = zip(self.height_list,
+                             self.depth_list,
+                             self.bh_radius_list,
+                             self.x_coords,
+                             self.y_coords,
+                             self.tilt_angle_list,
+                             self.orientation_angle_list)
+        return [Borehole(*cfg) for cfg in all_bh_configs]
+
+    def get_g_functions(self,
+                        alpha: float,
+                        time: List[float],
+                        options: Union[Dict[str, str], None] = None,
+                        solver_method: str = "equivalent",
+                        boundary_condition: str = "UHTR"):
+
+        """
+        Generates g-function values
+
+        Parameters
+        ----------
+        alpha: float
+            soil thermal diffusivity, in m^2/s
+        time: list[float]
+            time interval values for computing g-function values
+        options: Union[Dict[str, str], None]
+            Optional argument, options dict containing options for g-function computation
+        solver_method: str
+            optional argument, solver method for g-function computation. default: "equivalent".
+            other options: "similarities" or "detailed"
+        boundary_condition: str
+            optional argument, boundary condition for g-function computation, default: "UHTR"
+            other options: "UBWT" or "MFIT",
+
+        Returns
+        ----------
+        array of g-function values
+        """
+
+        if options is None:
+            options = {}
+
+        self.gfunc = gFunction(
+            self.as_config(),
+            alpha,
+            time=time,
+            boundary_condition=boundary_condition,
+            options=options,
+            method=solver_method,
+        )
+
+        return self.gfunc.gFunc
+
+    def to_list(self):
+        """returns a _list_ of g-function values"""
+        return self.gfunc.gFunc.tolist()
```

Comment:
@Myoldmopar OK, I have combined the classes. Maybe one final (for now) question, is regarding the return types for each of these.

`get_g_functions` returns a numpy array; `to_list` converts that array to a list. Do these seem OK for now?

Reply:
If it were my personal preference alone, I would say we just have get_g_functions return the plain list of floats and leave numpy out of it.  It's easy enough for an API client to create a numpy array of it if they want it.  It may be a slight inconvenience to a few people, but it's going to be completely insignificant in runtime and keeps up operating on plain python as much as possible.  And then we don't need this to_list function anymore.  

If we don't want to go that far, then I might still be inclined to remove this as_list function and just return a numpy array from get_g_functions.  The client can still do whatever they like with it.

Reply:
Make sense, hence my apprehension. I'll update that.